### PR TITLE
Add watchOS as a supported destination

### PR DESCRIPTION
**Pull request by tatsuky**
Hi, I'm new to this project. This PR adds the watchOS support for the `supportedDestinations` configuration, initially added in #1336.

## Changes

- Added the `watchOS` case to the `SupportedDestination` enum.
- Added a configuration template for watchOS.
- Added test cases.

## Tests

You can use the following example spec to verify the changes.

```yml
name: MyProject
options:
  bundleIdPrefix: com.myapp
targets:
  MyWatchKitApp:
    type: application
    platform: watchOS
    sources: [MyWatchKitApp]
    settings:
      MARKETING_VERSION: 1
      CURRENT_PROJECT_VERSION: 1
    dependencies:
      - target: MyFramework
    info:
      path: MyWatchKitApp/Info.plist
      properties:
        WKCompanionAppBundleIdentifier: com.myapp
        WKApplication: true
  MyFramework:
    type: framework
    supportedDestinations: [iOS, watchOS]
    platform: iOS
    sources: [MyFramework]
    settings:
      GENERATE_INFOPLIST_FILE: YES
```

Thanks.

### CODE REVIEW

File: Sources/ProjectSpec/SupportedDestination.swift [hunk 1]
```diff
@@ -40,6 +43,8 @@ extension SupportedDestination {
             return 3
         case .macCatalyst:
             return 4
+        case .watchOS:
+            return 5
```
Comment (tatsuky) [inline Sources/ProjectSpec/SupportedDestination.swift:47]: Hi @yonaskolb and @amatig,
I'm wondering why the priorities are currently in this order and where `watchOS` should be. Any thoughts?

Comment (tatsuky) [reply Sources/ProjectSpec/SupportedDestination.swift]: Adding test cases helped me realize this - `TARGETED_DEVICE_FAMILY` and `SUPPORTED_PLATFORMS` are ordered based on this priority. watchOS is `4`, meaning it needs to be between tvOS (`3`) and visionOS (`7`).

File: Tests/XcodeGenKitTests/ProjectGeneratorTests.swift [hunk 2]
```diff
@@ -484,7 +484,33 @@ class ProjectGeneratorTests: XCTestCase {
                 try expect(targetConfig1.buildSettings["ASSETCATALOG_COMPILER_APPICON_NAME"] as? String) == "AppIcon"
                 try expect(targetConfig1.buildSettings["CODE_SIGN_IDENTITY"] as? String) == "iPhone Developer"
             }
-            
+
+            $0.it("supportedDestinations merges settings - iOS, watchOS") {
```
Comment (giginet) [inline Tests/XcodeGenKitTests/ProjectGeneratorTests.swift:488]: These tests can be parameterized tests to reduce redundancy.

But it's better existing test cases also become parameterized tests.
So I think it's good at this time.

**Review by giginet — APPROVED** [review]
Great contribution! I wished this feature 💯 

I'll test this feature for my use case.

**Review by PierreJanineh — APPROVED** [review]

**Review by yonaskolb — APPROVED** [review]
Thank you @tatsuky

**Pull request merged**